pkg/controller/kvcache/kvcache_controller.go

```diff
@@ -463,6 +465,52 @@ func (r *KVCacheReconciler) reconcileDeployment(ctx context.Context, kvCache *or
 		affinity.PodAffinity = podAffinity
 	}
 
+	// Add pod anti-affinity handling
+	if value, ok := kvCache.Annotations[KVCacheAnnotationPodAntiAffinityKey]; ok {
+		podAntiAffinity := &corev1.PodAntiAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+				{
+					LabelSelector: &metav1.LabelSelector{
+						MatchExpressions: []metav1.LabelSelectorRequirement{
+							{
+								Key:      "model.aibrix.ai/name",
```

Comment:
what's the usage here? do you want to apply anti-affinity to other engine pods or just kv cache pod? 

the cache pod doesn't have `model.aibrix.ai/name` labels, right?